From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Thu, 27 Aug 2026 11:38:35 +0200
Subject: fix(znet): support mktemp applet from busybox in module-setup.sh

busybox's `mktemp` applet does not support the long form `--tmpdir` but
it supports the short form `-p`.

Part of: https://github.com/dracut-ng/dracut/issues/2437
Origin: upstream, https://github.com/dracut-ng/dracut/commit/dcbe20e146b570ac3bff80bbebccd2fca973879e
---
 modules.d/74znet/module-setup.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/74znet/module-setup.sh b/modules.d/74znet/module-setup.sh
index 6c340a6..55951e7 100755
--- a/modules.d/74znet/module-setup.sh
+++ b/modules.d/74znet/module-setup.sh
@@ -26,7 +26,7 @@ install() {
     inst_multiple grep sed seq readlink chzdev
     if [[ $hostonly ]]; then
         local _tempfile
-        _tempfile=$(mktemp --tmpdir="${DRACUT_TMPDIR}" dracut-zdev.XXXXXX)
+        _tempfile=$(mktemp -p "${DRACUT_TMPDIR}" dracut-zdev.XXXXXX)
         {
             chzdev qeth --export - --configured --persistent --quiet --type
             chzdev lcs --export - --configured --persistent --quiet --type
